controllers/vrg_volrep.go

```diff
@@ -1622,30 +1672,31 @@ func (v *VRGInstance) fetchAndRestorePV(result *ctrl.Result) error {
 			continue
 		}
 
-		var pvList []corev1.PersistentVolume
+		var pvCount, pvcCount int
 
-		pvList, err = downloadPVs(objectStore, v.s3KeyPrefix())
+		// Restore all PVs found in the s3 store. If any failure, the next profile will be retried
+		pvCount, err = v.restorePVsFromObjectStore(objectStore, s3ProfileName)
 		if err != nil {
-			v.log.Error(err, fmt.Sprintf("error fetching PV cluster data from S3 profile %s", s3ProfileName))
-
 			continue
 		}
 
-		v.log.Info("Found PVs in s3 store", "count", len(pvList), "profile", s3ProfileName)
+		// Attempt to restore all PVCs from this profile. If a PVC is missing from this s3 stores or fails
+		// to restore, there will be a warning, but not a failure (no retry). In such cases, the PVC may be
+		// created when the application is created and it will bind to the PV correctly if the PVC name
+		// matches the PV.Spec.ClaimRef. However, the downside of tolerating failure is if an operator like
+		// CrunchyDB is responsible for creating and managing the lifecycle of their own PVCs, a newly created
+		// PVC may cause a new PV to be created.
+		// Ignoring PVC restore errors helps with the upgrade from ODF-4.12.x to 4.13
+		pvcCount, err = v.restorePVCsFromObjectStore(objectStore, s3ProfileName)
 
-		if err = v.checkPVClusterData(pvList); err != nil {
-			errMsg := fmt.Sprintf("Error found in PV cluster data in S3 store %s", s3ProfileName)
-			v.log.Info(errMsg)
-			v.log.Error(err, fmt.Sprintf("Resolve PV conflict in the S3 store %s to deploy the application", s3ProfileName))
-
-			return fmt.Errorf("%s: %w", errMsg, err)
-		}
+		if err != nil || pvCount != pvcCount {
```

Comment:
(future) For backward compat, we should find no PVCs or all PVCs, there should be no need to handle an intermediate state. So that we are sure we performed a full PVC/PV backup/restore in this case.

Reply:
issue created here: https://github.com/RamenDR/ramen/issues/834